src/trackers/classes/site.js

```diff
@@ -145,8 +161,31 @@ async function _processRequest (requestData, site) {
     if (!site.isFirstParty(request.url)) {
         const nameservers = await shared.nameservers.resolveNs(request.domain)
 
-        if (nameservers) {
+        if (nameservers && nameservers.length) {
             request.nameservers = nameservers
+
+            // option to group by nameservers is set in the config
+            if (shared.nameserverMap) {
+                let nsMatch
+
+                shared.nameserverMap.some(nsEntry => {
```

Comment:
1. Using `some` without a return is a nice trick, but I wonder if a for loop with `break` would show the intention more clearly.
 2. `nameserverMap` is an array?? Could it be named better?
 3. We're iterating the whole `nameserverMap` for every request (in the worst case). If the length of `nameserverMap` increases then complexity is significantly increased. Is it worth having a simple reverse lookup to make this constant time?

Reply:
1. 👍 
2. 👍 renamed to nameserverList
3. 👍 added a reverse lookup and checking that one of the nameservers is in the lookup first